tests/test_blackoilfluidsystem_nonstatic.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+  Copyright 2025 Equinor ASA
+
+  This file is part of the Open Porous Media project (OPM).
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#define BOOST_TEST_MODULE TestBlackOilFluidSystemNonStatic
+#include <config.h>
+
+#include <cstddef>
+
+#include <boost/test/unit_test.hpp>
+
+#include <opm/material/fluidsystems/BlackOilFluidSystem.hpp>
+#include <opm/material/fluidsystems/BlackOilFluidSystemNonStatic.hpp>
+
+BOOST_AUTO_TEST_CASE(TestThrowOnUnInitialized)
```

Comment:
The amount of tests is insufficient. Please add at least one test (can use test_eclblackoilfluidsystem.cpp as a starting point) where several properties are evaluated at many points with both static and nonstatic systems, and exact equality is tested for.